openpype/hosts/houdini/plugins/publish/collect_rop_frame_range.py

```diff
@@ -2,40 +2,103 @@
 """Collector plugin for frames data on ROP instances."""
 import hou  # noqa
 import pyblish.api
+from openpype.lib import BoolDef
 from openpype.hosts.houdini.api import lib
+from openpype.pipeline import OpenPypePyblishPluginMixin
 
 
-class CollectRopFrameRange(pyblish.api.InstancePlugin):
+class CollectRopFrameRange(pyblish.api.InstancePlugin,
+                           OpenPypePyblishPluginMixin):
+
     """Collect all frames which would be saved from the ROP nodes"""
 
+    hosts = ["houdini"]
     order = pyblish.api.CollectorOrder
     label = "Collect RopNode Frame Range"
+    use_asset_handles = True
 
     def process(self, instance):
 
         node_path = instance.data.get("instance_node")
         if node_path is None:
             # Instance without instance node like a workfile instance
+            self.log.debug(
+                "No instance node found for instance: {}".format(instance)
+            )
             return
 
         ropnode = hou.node(node_path)
-        frame_data = lib.get_frame_data(ropnode)
 
-        if "frameStart" in frame_data and "frameEnd" in frame_data:
+        attr_values = self.get_attr_values_from_data(instance.data)
 
-            # Log artist friendly message about the collected frame range
-            message = (
-                "Frame range {0[frameStart]} - {0[frameEnd]}"
-            ).format(frame_data)
-            if frame_data.get("step", 1.0) != 1.0:
-                message += " with step {0[step]}".format(frame_data)
-            self.log.info(message)
+        if attr_values.get("use_handles", self.use_asset_handles):
+            asset_data = instance.context.data["assetEntity"]["data"]
+            handle_start = asset_data.get("handleStart", 0)
+            handle_end = asset_data.get("handleEnd", 0)
+        else:
+            handle_start = 0
+            handle_end = 0
 
-            instance.data.update(frame_data)
+        frame_data = lib.get_frame_data(
+            ropnode, handle_start, handle_end, self.log
+        )
+
+        if not frame_data:
+            return
 
-            # Add frame range to label if the instance has a frame range.
-            label = instance.data.get("label", instance.data["name"])
-            instance.data["label"] = (
-                "{0} [{1[frameStart]} - {1[frameEnd]}]".format(label,
-                                                               frame_data)
+        # Log artist friendly message about the collected frame range
+        frame_start = frame_data["frameStart"]
+        frame_end = frame_data["frameEnd"]
+
+        if attr_values.get("use_handles"):
```

Comment:
Shouldn't this fallback to default value `use_asset_handles`?
```suggestion
        if attr_values.get("use_handles", self.use_asset_handles):
```